packages/stores/types.ts

```diff
@@ -548,6 +548,13 @@ type TContractStore = {
     contract_update_take_profit: string;
     has_contract_update_stop_loss: boolean;
     has_contract_update_take_profit: boolean;
+    last_contract: {
```

Comment:
@henry-deriv I don't really understand why this TContractStore type includes a mix of properties from 2 stores: contract-store and contract-trade-store. last_contract doesn't exist in contract-store, and has_contract_update_stop_loss doesn't exist in contract-trade-store. i will be also changing this in my PR, it would be correct to separate them. what do you think?

Reply:
` onMount(contract_id) {
        if (contract_id) {
            this.contract_id = contract_id;
            this.contract_store = new ContractStore(this.root_store, { contract_id });
            this.subscribeProposalOpenContract();
            WS.setOnReconnect(() => {
                if (!this.root_store.client.is_switching) {
                    this.subscribeProposalOpenContract();
                }
            });
        }
    }`
    
    more specifically `this.contract_store = new ContractStore(this.root_store, { contract_id });`

Reply:
contract-replay-store is essentially its open props plus the whole `contractstore`

Reply:
we also have this in contract-trade-store

`const contract = new ContractStore(this.root_store, { contract_id });`

`this.contracts.push(contract);`

Reply:
@henry-deriv i don't mean contract-replay-store :) there is contract-store which is the same as contract you are talking about, and there is a separate contract-trade-store, and also there is the 3rd store - contract-replay-store. please take a look at my changes here https://github.com/kate-deriv/deriv-app/pull/22/files🙏🏻

Reply:
the rest LGTM, @kate-deriv do you think we can merge this for now because i'll change it in my PR anyway ? [https://github.com/kate-deriv/deriv-app/pull/22/files](https://github.com/kate-deriv/deriv-app/pull/22/files%F0%9F%99%8F%F0%9F%8F%BB)

Reply:
yeah i was just saying contract-replay-store and contract-trade-store have instantiations of contract-store.

And thats the reason why we have mix of properties/states in all three stores.

Reply:
If you don't mind it will be better to merge Henry's branch now and then fix it in your PR in order to avoid conflicts

Reply:
`I don't really understand why this TContractStore type includes a mix of properties from 2 stores: contract-store and contract-trade-store. last_contract doesn't exist in contract-store, and has_contract_update_stop_loss doesn't exist in contract-trade-store. i will be also changing this in my PR, it would be correct to separate them. what do you think?`

Answer to this whoever put it in was probably confused 😅

Reply:
Yes lets separate them and type them properly 👍